### PR TITLE
[MRG] Fixed error when running create_ecg_epochs

### LINKED ISSUE #4137: create_ecg_epochs crashes if data is not preloaded
I found this error when I ran create_ecg_epochs before create_eog_epochs. 

create_eog_epochs will preload the data when it creates the epochs, and then create_ecg_epochs will run fine. 
However, create_ecg_epochs will crash on data that is not preloaded, if run first. This happens when it tries to run the line `raw.add_channels([ecg_raw])` which needs the data to be preloaded.

You can easily recreate this error by running create_ecg_epochs on raw_fif data that has not been preloaded. I can create an example if need be, I can also create a pull request to fix this if you want me to. My solution to the problem would just be preloading the data though, not sure if that is too naive of an approach or not.

### PR DESCRIPTION
**Pull request by ARudiuk**
-Raw fif data will be preloaded when running create_ecg_epochs
-Added test that checks if create_ecg_epochs works with data
that is not preloaded

closes #4137

### CODE REVIEW

File: mne/preprocessing/tests/test_ecg.py [hunk 1]
```diff
@@ -55,4 +55,13 @@ def test_find_ecg():
...
     create_ecg_epochs(raw)
 
+
+def test_ecg_preload():
+    # Test if ECG analysis will work on data that is not preloaded
+    raw = read_raw_fif(raw_fname, preload=False)
+    # There should be no ECG channels, or else preloading will not be
+    # tested
+    assert_true('ecg' not in raw)
+    create_ecg_epochs(raw)
```
Comment (agramfort) [inline mne/preprocessing/tests/test_ecg.py]: can you rather update a test or make sure it's crazy fast to run by cropping a lot?

Comment (ARudiuk) [inline mne/preprocessing/tests/test_ecg.py]: I updated the existing test to check the new functionality by not preloading the data. Let me know if that works

File: mne/preprocessing/tests/test_ecg.py [hunk 2]
```diff
@@ -16,7 +16,8 @@
 
 def test_find_ecg():
     """Test find ECG peaks."""
-    raw = read_raw_fif(raw_fname)
+    # Test if ECG analysis will work on data that is not preloaded
+    raw = read_raw_fif(raw_fname, preload=False)
```
Comment (agramfort) [inline mne/preprocessing/tests/test_ecg.py]: preload is already false by default

**Comment by agramfort** [comment]
@ARudiuk can you see why travis complains?

**Comment by ARudiuk** [comment]
@agramfort the error is due to a spelling mistake in the decoding module it seems

> mne/decoding/base.py:154: probablistic  ==> probabilistic
> make[1]: *** [codespell-error] Error 1
> make[1]: Leaving directory `/home/travis/build/mne-tools/mne-python'
> make: *** [pep] Error 2

Do you want me to go fix it in this pull request?

File: mne/preprocessing/tests/test_ecg.py (hunk 2 above)
Comment (ARudiuk) [inline mne/preprocessing/tests/test_ecg.py]: True, but my worry is that if the preload default is changed one day, then the test will not pick up if `create_ecg_epochs` fails to load the data.

Do you still want me to make the change?

Comment (Eric89GXL) [inline mne/preprocessing/tests/test_ecg.py]: I think it's fine to be explicit like this

**Comment by Eric89GXL** [comment]
> Do you want me to go fix it in this pull request?

I fixed it in `master` and restarted the build here

**Comment by agramfort** [comment]
thx @ARudiuk

**Comment by ARudiuk** [comment]
np!

**Pull request merged**